### DISCUSSION TITLE
build error: github.com/tecbot/gorocksdb: invalid flag in pkg-config --cflags: -faligned-new

### DISCUSSION THREAD

**Discussion by chen982** [milvus-io/milvus#27914] [Q&A and General discussion]
When I try to build the milvus , it shows the error:
Build Tag: v2.2.12-dev
Build Time: Wed Oct 25 06:33:16 UTC 2023
Git Commit: 1f1ac8f10
Go Version: go version go1.20.7 linux/amd64
Building Milvus ...
github.com/tecbot/gorocksdb: invalid flag in pkg-config --cflags: -faligned-new
Makefile:29: recipe for target 'milvus' failed
make: *** [milvus] Error 1
How to fix it?

**Comment by chen982** [comment]
When I try to export CGO_CFLAGS_ALLOW='-faligned-new' before make:
It shows another error like this:
cc1: warning: command line option ‘-std=c++11’ is valid for C++/ObjC++ but not for C
cc1: warning: command line option ‘-faligned-new=1’ is valid for C++/ObjC++ but not for C
# github.com/tecbot/gorocksdb
../gowork/pkg/mod/github.com/milvus-io/gorocksdb@v0.0.0-20220624081344-8c5f4212846b/db.go:620:3: too many arguments in call to (_Cfunc_rocksdb_approximate_sizes)
        have (*_Ctype_struct_rocksdb_t, _Ctype_int, **_Ctype_char, *_Ctype_ulong, **_Ctype_char, *_Ctype_ulong, *_Ctype_ulong, **_Ctype_char)
        want (*_Ctype_struct_rocksdb_t, _Ctype_int, **_Ctype_char, *_Ctype_ulong, **_Ctype_char, *_Ctype_ulong, *_Ctype_ulong)
../gowork/pkg/mod/github.com/milvus-io/gorocksdb@v0.0.0-20220624081344-8c5f4212846b/db.go:666:3: too many arguments in call to (_Cfunc_rocksdb_approximate_sizes_cf)
        have (*_Ctype_struct_rocksdb_t, *_Ctype_struct_rocksdb_column_family_handle_t, _Ctype_int, **_Ctype_char, *_Ctype_ulong, **_Ctype_char, *_Ctype_ulong, *_Ctype_ulong, **_Ctype_char)
        want (*_Ctype_struct_rocksdb_t, *_Ctype_struct_rocksdb_column_family_handle_t, _Ctype_int, **_Ctype_char, *_Ctype_ulong, **_Ctype_char, *_Ctype_ulong, *_Ctype_ulong)
../gowork/pkg/mod/github.com/milvus-io/gorocksdb@v0.0.0-20220624081344-8c5f4212846b/filter_policy.go:49:67: cannot use _Ctype_double(bitsPerKey) (value of type _Ctype_double) as _Ctype_int value in argument to (_Cfunc_rocksdb_filterpolicy_create_bloom)
../gowork/pkg/mod/github.com/milvus-io/gorocksdb@v0.0.0-20220624081344-8c5f4212846b/filter_policy.go:55:72: cannot use _Ctype_double(bitsPerKey) (value of type _Ctype_double) as _Ctype_int value in argument to (_Cfunc_rocksdb_filterpolicy_create_bloom_full)
Makefile:29: recipe for target 'milvus' failed
make: *** [milvus] Error 1

**Comment by yhmo** [comment]
Not sure why you got these errors, different machines have different environments, it is hard to find the reason.

Typically, we build milvus source by the following steps:
1. make sure the os meets the requirement
Ubuntu 18.04 or later
MacOS 12.0 or later

2. make sure the go/cmake/gcc version meet the requirement
```
go: >= 1.18
cmake: >= 3.18
gcc: 7.5
```
3. install the dependencies
`./scripts/install_deps.sh`

4. make

**Comment by xiaofan-luan** [comment]
hit similar issue

**Reply by xiaofan-luan** [reply]
To solve the issue, you will need to setup your PKG_CONFIG_PATH and some other system configs.


      export PKG_CONFIG_PATH="${PKG_CONFIG_PATH}:$ROOT_DIR/internal/core/output/lib/pkgconfig"
      export DYLD_LIBRARY_PATH=$ROOT_DIR/internal/core/output/lib
      export RPATH=$DYLD_LIBRARY_PATH;;

**Reply by xiaofan-luan** [reply]
I set this on my goland configuration and it works

**Reply by xiaofan-luan** [reply]
run setenv.sh should  work as well
